### PR TITLE
fix the outline style lost when overriding TextMeshPro Font

### PR DESCRIPTION
**Pull request by ekibun**
The outine style defined in material will lost when overriding font. This pr is going to copy the style to a new material.

### CODE REVIEW

File: src/XUnity.AutoTranslator.Plugin.Core/TextTranslationInfo.cs [hunk 1]
```diff
@@ -132,14 +133,47 @@ public void ChangeFont( object ui )
...
+
+               if( oldMaterial != null && newMaterial != null )
+               {
+                  object copyMaterial = null;
+                  if( !_matCopies.TryGetValue( oldMaterial, out copyMaterial ) )
+                  {
+                     copyMaterial = _matCopies[ oldMaterial ] = UnityEngine.Object.Instantiate( oldMaterial as Material );
+
+                     // keep original font
+                     var uiCopy = UnityEngine.Object.Instantiate( ui as UnityEngine.Object );
```
Comment (ManlyMarco) [inline src/XUnity.AutoTranslator.Plugin.Core/TextTranslationInfo.cs:151]: This copies the original text mesh component onto a new game object, right? Why is this necessary?

Comment (ekibun) [reply src/XUnity.AutoTranslator.Plugin.Core/TextTranslationInfo.cs:151]: the textmesh wil be removed if the material do not refered to a ui object, it may cause the overried font cannot be triggered back by pressing ctrl+f.
i haven't found a better way to keep that :(

Comment (ManlyMarco) [reply src/XUnity.AutoTranslator.Plugin.Core/TextTranslationInfo.cs:151]: The font and material should stay alive if they are referenced by any alive component in the scene, I'd guess adding an unity object list to the plugin class would work, maybe even as a static field? More info [here](https://docs.unity3d.com/530/Documentation/ScriptReference/EditorUtility.UnloadUnusedAssetsImmediate.html) and [here](https://discussions.unity.com/t/unloadunusedassets-what-exactly-does-unused-mean/131720/3).

**Review by ManlyMarco — APPROVED** [review]
